applications/HDF5Application/python_scripts/core/operations/model_part.py

```diff
@@ -28,267 +35,286 @@ def Prefix(pattern, model_part, time_format=''):
     return prefix
 
 
-class ModelPartInput:
-    '''Reads a model part from a file.'''
+class IOOperation(KratosMultiphysics.Operation): # IOOperation(KratosMultiphysics.Operation, metaclass = abc.ABCMeta) # <== conflicting metaclasses
+    """ @brief Base class for HDF5 IO operations."""
 
-    def __init__(self, settings):
-        settings.SetDefault('prefix', '/ModelData')
-        self.prefix = settings['prefix']
-        if '<time>' in self.prefix:
-            settings.SetDefault('time_format', '0.4f')
-            self.time_format = settings['time_format']
+    def __init__(self,
+                 model_part: KratosMultiphysics.ModelPart,
+                 parameters: KratosMultiphysics.Parameters,
+                 file: KratosHDF5.HDF5File):
+        super().__init__()
+        self.__model_part = model_part
+        self.__file = file
+        self.__time_format = parameters["time_format"].GetString()
 
-    def __call__(self, model_part, hdf5_file):
-        if hasattr(self, 'time_format'):
-            prefix = Prefix(self.prefix, model_part, self.time_format)
-        else:
-            prefix = Prefix(self.prefix, model_part)
-        KratosHDF5.HDF5ModelPartIO(
-            hdf5_file, prefix).ReadModelPart(model_part)
+        # Pick necessary entries from the input parameters
+        # (assuming they're all present)
+        self.__parameters = KratosMultiphysics.Parameters()
+        self.__parameters.AddString("prefix", Prefix(parameters["prefix"].GetString(), self.model_part, self.time_format))
 
+    #@abc.abstractmethod # <== missing the ABCMeta metaclass
+    def Execute(self) -> None:
+        raise RuntimeError("Call to a pure abstract method")
```

Comment:
```suggestion
        raise NotImplementedError("Call to a pure abstract method")
```

Reply:
This nicely works with VSCode, and it does not grey out the rest of the code after calling this if it is throwing `NotImplementedError`, otherwise, it greys out the rest of the code after a call to this method.